tools/fuzzer.cpp

```diff
@@ -0,0 +1,211 @@
+          generateVelocypack(builder, depth + 1, randomGenerator);
+        }
+        builder.close();
+        break;
+      }
+      case ADD_BOOLEAN:
+        builder.add(Value(randomGenerator.mt64() % 2 ? true : false));
+        break;
+      case ADD_STRING:
+        addString(builder, randomGenerator);
+        break;
+      case ADD_NULL:
+        builder.add(Value(ValueType::Null));
+        break;
+      case ADD_UINT64:
+        builder.add(Value(randomGenerator.mt64()));
+        break;
+      case ADD_INT64: {
+        uint64_t uintValue = randomGenerator.mt64();
+        int64_t intValue;
+        memcpy(&intValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(intValue));
+        break;
+      }
+      case ADD_DOUBLE: {
+        uint64_t uintValue = randomGenerator.mt64();
+        double doubleValue;
+        memcpy(&doubleValue, &uintValue, sizeof(uintValue));
+        builder.add(Value(doubleValue));
+      }
+      default:
+        break;
+    }
+    break;
+  }
+}
+
+int main(int argc, char* argv[]) {
+  VELOCYPACK_GLOBAL_EXCEPTION_TRY
+    std::vector<std::thread> threads;
+    bool isTypeAssigned = false;
+    size_t numIterations = 1;
+    size_t numThreads = 1;
+    Format format;
+
+    int i = 1;
+    while (i < argc) {
+      char const *p = argv[i];
+      if (isOption(p, "--help")) {
+        usage(argv);
+        return EXIT_SUCCESS;
+      } else if (isOption(p, "--vpack") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::VPACK;
+      } else if (isOption(p, "--json") && !isTypeAssigned) {
+        isTypeAssigned = true;
+        format = Format::JSON;
+      } else if (int value = atoi(p); value >= 1) {
+        if (i == 3) { // as threads and iterations are optional arguments, threads must be preceeded by iterations
+                      // in the command line, hence, argument 3
+          numThreads = value;
+        } else {
+          numIterations = value;
+        }
+      } else {
+        usage(argv);
+        return EXIT_FAILURE;
+      }
+      ++i;
+    }
+
+    size_t itsPerThread = numIterations / numThreads;
+    size_t leftoverIts = numIterations % numThreads;
+
+    auto threadCallback = [format] (size_t iterations) {
+      RandomGenerator randomGenerator;
+      while (iterations-- > 0) {
+        Builder builder;
+        generateVelocypack(builder, 0, randomGenerator);
+
+        if (format == Format::JSON) {
+          Parser parser;
+          parser.parse(builder.slice().toJson());
+        } else {
+          Validator validator;
+          validator.validate(builder.slice().start(), builder.slice().byteSize());
+        }
+      }
+    };
+
+    for (size_t i = 0; i < numThreads; ++i) {
+      size_t iterations = itsPerThread;
+      if (i == numThreads - 1) {
+        iterations += leftoverIts;
+      }
+      threads.emplace_back(std::thread(threadCallback, iterations));
+    }
+
+    for (auto &thread: threads) {
```

Comment:
Coding guidelines suggest:
```suggestion
    for (auto& thread: threads) {
```